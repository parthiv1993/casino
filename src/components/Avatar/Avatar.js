import PropTypes from 'prop-types';
import { baseURL } from '../../api/urls';
import { AvatarImage } from './Avatar.styles';

const Avatar = ({ avatarLink }) => {
  const imageLink = `${baseURL}/${avatarLink}`;
  return <AvatarImage src={imageLink} alt="avatar" />;
};

Avatar.propTypes = {
  avatarLink: PropTypes.string
};

export default Avatar;
