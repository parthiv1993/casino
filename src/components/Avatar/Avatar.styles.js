import styled from 'styled-components';

export const AvatarImage = styled.img.attrs({
  height: 48,
  width: 48
})`
  border-radius: 50%;
`;
