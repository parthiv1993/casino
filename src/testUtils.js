import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
