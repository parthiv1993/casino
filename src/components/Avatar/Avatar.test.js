import { renderComponentWithLightTheme } from '../../testUtils';
import Avatar from './Avatar';

describe('Avatar', () => {
  it('should match snapshot', () => {
    const { container } = renderComponentWithLightTheme(<Avatar avatarLink="fa-username" />);
    expect(container).toMatchSnapshot();
  });
});
