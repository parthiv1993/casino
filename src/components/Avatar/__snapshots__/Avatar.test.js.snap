// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
.c0 {
  border-radius: 50%;
}

<div>
  <img
    alt="avatar"
    class="c0"
    height="48"
    src="http://localhost:3001/fa-username"
    width="48"
  />
</div>
`;
